### DISCUSSION TITLE
Combine REST API data source and Table

### DISCUSSION THREAD

**Discussion by uk94** [Budibase/budibase#4513] [Help]
Hi,

I have a REST API source that queries some website data (disk used, disk quota etc.), and on another table, I have all my websites defined. 

Both REST API data source and table have a column called domain.

I want to combine both data sources and tables.

Is this possible?

Thanks.

**Comment by mjashanks** [comment]
Hi @u44khn  - it depends on what you mean by "Combine"

What you could do, on screen is

```
> Data provider : REST API
   > Repeater 
       > Data provider : Table - filter where domain = RESTAPI.domain
```

This method might performance issues if your rest API returns an array with a lot of items. I.e. it will mean that one `Table` request will get send for every array item.

Another method that you could do is

```
> Data provider : REST API 
  > Data provider : Table - filter where domain = RESTAPI.domain
      > Repeater (REST API)
          > Components with Javascript to map table rows and REST rows
```

in this method, the Table and REST request each only happens once, but you have to write javascript to to something like

```javascript
const tableRow = $("Table Data Provider.Rows").find(r => r.domain === $("REST Repeater.REST.domain"));
return tableRow; // and do something with the table row
```

I've just pulled this from my head, so it's not checked! But something like that should work.
